include/RAJA/config.hpp.in

```diff
@@ -436,12 +436,28 @@ const int DATA_ALIGN = @RAJA_DATA_ALIGN@;
 #define RAJA_MAX_ALIGN 16
 #define RAJA_FORCEINLINE_RECURSIVE
 #define RAJA_INLINE inline  __attribute__((always_inline))
-#if !defined(__NVCC__)
+
+#if !defined(__NVCC__) // purely GCC, CUDA compiler not active
 #define RAJA_UNROLL RAJA_PRAGMA(GCC unroll 10000)
 #define RAJA_UNROLL_COUNT(N) RAJA_PRAGMA(GCC unroll N)
-#else
-#define RAJA_UNROLL RAJA_PRAGMA(unroll)
-#define RAJA_UNROLL_COUNT(N) RAJA_PRAGMA(unroll N)
+#else // CUDA compiler active, with GCC host compiler
+#if defined(__NVCC_DIAG_PRAGMA_SUPPORT__) // CUDA >= 11.5
+#define RAJA_UNROLL \
+        _Pragma ("nv_diagnostic push") \
+        _Pragma ("nv_diag_suppress 1675") \
+        _Pragma ("GCC unroll 10000") \
+        _Pragma ("nv_diagnostic pop")
+#define RAJA_UNROLL_COUNT(N) \
+        _Pragma ("nv_diagnostic push") \
+        _Pragma ("nv_diag_suppress 1675") \
+        _Pragma ("GCC unroll N") \
```

Comment:
Does it work to use N in a string like this?

Reply:
I think `_Pragma ("GCC unroll " #N)` should work.

Reply:
Should be fixed now. Had to use the RAJA stringified macros.